lib/lotus/commands/generate/abstract.rb

```diff
@@ -0,0 +1,75 @@
+require 'lotus/environment'
+require 'thor'
+
+module Lotus
+  module Commands
+    class Generate < Thor
+      class Abstract
+        VALID_TEST_TYPES = ['minitest', 'rspec'].freeze
+
+        attr_reader :options, :base_path
+
+        def initialize(options)
+          @options = Lotus::Utils::Hash.new(options).symbolize!
+          assert_options!
+          # TODO: find a good name for this...
+          @base_path = Pathname.pwd
+        end
+
+        def start
+          raise NotImplementedError
+        end
+
+        private
+
+        def lotusrc_options
+          @lotusrc_options ||= Lotusrc.new(Pathname.new(base_path)).read
+        end
+
+        def environment
+          @environment ||= Lotus::Environment.new(options)
+        end
+
+        def template_source_path
+          # TODO: how to do this clean...
+          generator = self.class.name.split('::').last.downcase
+          Pathname.new(::File.dirname(__FILE__) + "/../../generators/#{generator}/").realpath
+        end
+
+        # @since x.x.x
+        # @api private
+        def test_type
+          options.fetch(:test, default_test_type)
+        end
+
+        def default_test_type
+          lotusrc_options.fetch(:test)
+        end
+
+        def template_engine
+          options.fetch(:template, default_template_engine)
+        end
+
+        def default_template_engine
+          lotusrc_options.fetch(:template)
+        end
+
+        # @since x.x.x
+        # @api private
+        def assert_test_type!
```

Comment:
I took over the _assert*_ naming convention from previous code. Personally i use _ensure*_ but also  _assert*_ seems quite common.

It's also quite common to append bang, when the method is known to raise an exception in case of a problem. You can see it throughout ActiveRecord and other gems.
Or as Matz puts it: more dangerous than the non-bang counterpart. Which of course is a moot point because the non-bang-counterpart does not exist.
What do others say? What is the best practice in Lotus?